core/services/workflows/syncer/fetcher.go

```diff
@@ -23,17 +24,45 @@ type FetcherService struct {
 	lggr    logger.Logger
 	och     *webapi.OutgoingConnectorHandler
 	wrapper gatewayConnector
+	limits  *ArtifactConfig
 }
 
+type ArtifactConfig struct {
```

Comment:
Hmmm -- it looks like you're implementing this check over two layers the fetcher + the syncer

Did you consider instead:
- passing through the config to the handler
- then in each call to the Fetcher setting the config setting directly as a static value

That way the changes will be more cohesive -- the Fetcher doesn't know anything about artifact types, that's strictly a concern of the handler